### PR TITLE
feat(argo-cd): Allow modification of path and port for dex liveness and readiness probes

### PR DESCRIPTION
**Pull request by fboula**
Allow to change the default http path and port for Dex server.

Fixes: https://github.com/argoproj/argo-cd/issues/9091

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/templates/dex/deployment.yaml [hunk 1]
```diff
@@ -99,8 +99,8 @@ spec:
         {{- if .Values.dex.livenessProbe.enabled }}
         livenessProbe:
           httpGet:
-            path: /healthz/live
-            port: metrics
+            path: {{ .Values.dex.livenessProbe.httpPort }}
+            port: {{ .Values.dex.livenessProbe.httpPath }}
```
Comment (pdrastil) [inline charts/argo-cd/templates/dex/deployment.yaml:103]: The port name is hardcoded in deployment. This cannot be configured.

Comment (pdrastil) [reply charts/argo-cd/templates/dex/deployment.yaml:103]: Same applies most likely to path. Configuration is automatically generated by init container here.
Ref: https://github.com/argoproj/argo-cd/blob/4dc91dcb9df77071d0d8a83511449e2297d984b1/util/dex/config.go#L44C10-L44C19

Comment (fboula) [reply charts/argo-cd/templates/dex/deployment.yaml:103]: I understand your concern. This is not an ideal solution but for users relying on Google Cloud OIDC, the solution proposed in https://github.com/argoproj/argo-cd/issues/9091 seems to be a viable way to ensure dex server is restarted when it is not able to open the google connector. 

The default values are not introducing any behaviour changes so only chart users specifically changing the proposed values would see a difference.

The change would help, I understand some users may compromise their deployment if they configure a wrong value here, but if I can provide some documentation of the possible values or with a helm template function to ensure the set port is in the list of existing ports

WDYT ?

Comment (pdrastil) [reply charts/argo-cd/templates/dex/deployment.yaml:103]: Hi, thanks for providing context about this change. I think it's ok to allow configuration for this edge case.

**Review by pdrastil — APPROVED** [review]

**Comment by fboula** [comment]
I had to rebase my branch, let me know if the semver is ok.

**Review by mkilchhofer — APPROVED** [review]

**Review by pdrastil — APPROVED** [review]

**Pull request merged**